src/metabase/models/model_index.clj

```diff
@@ -207,7 +209,10 @@
                   :model-index-id  :model_index.id}
    :joins        {:model_index [:model/ModelIndex [:= :model_index.id :this.model_index_id]]
                   :model       [:model/Card [:= :model.id :model_index.model_id]]
-                  :collection  [:model/Collection [:= :collection.id :model.collection_id]]}})
+                  :collection  [:model/Collection [:= :collection.id :model.collection_id]]}
+   :update-cond  [:and
```

Comment:
Any thoughts on a better name here?

Reply:
I first have to come to the terms with needing this.

Do I understand right that these entries have a compound key? Does this mean that the way we use `:model_pk` as the `:model_id` (see `:attrs` above) is wrong?

Reply:
As discussed, yeah we need to update spec to handle compound ids holistically, then we won't need this property.

Since we're only using half the logical id right now, we're losing lots of data.

This explains why we needed to add that `distinct` in the ingestion a while back.